MDANSE/Src/MDANSE/Mathematics/Geometry.py

```diff
@@ -110,51 +115,63 @@ def moment_of_inertia(
+def generate_sphere_points(n_samples: int) -> np.ndarray:
     """Returns list of 3d coordinates of points on a sphere using the
     Golden Section Spiral algorithm.
-    """
-
-    inputs = np.arange(int(n))
-    points = np.empty([len(inputs), 3])
-
-    inc = np.pi * (3 - np.sqrt(5))
 
-    offset = 2 / float(n)
-
-    y = inputs * offset - 1 + (offset / 2)
-    r = np.sqrt(1 - y * y)
-    phi = inputs * inc
-    points[:, 0] = np.cos(phi) * r
-    points[:, 1] = y
-    points[:, 2] = np.sin(phi) * r
-
-    return points
+    Returns coordinates on a sphere with radius=1 and centre=0,0,0.
 
+    Parameters
+    ----------
+    n_samples : int, optional
+        number of points on the sphere, by default 1000
 
-def random_points_on_sphere(radius=1.0, nPoints=100):
-    points = np.zeros((3, nPoints), dtype=np.float64)
+    Returns
+    -------
+    npt.NDArray[float]
+        (n_samples, 3) array of point coordinates on a sphere
+    """
 
-    theta = 2.0 * np.pi * np.random.uniform(nPoints)
-    u = np.random.uniform(-1.0, 1.0, nPoints)
-    points[0, :] = radius * np.sqrt(1 - u**2) * np.cos(theta)
-    points[1, :] = radius * np.sqrt(1 - u**2) * np.sin(theta)
-    points[2, :] = radius * u
+    indices = np.arange(n_samples)
+    angle_step = np.pi * (np.sqrt(5.0) - 1.0)  # angle step in radians
 
-    return points
+    ys = 1 - (indices / (n_samples - 1)) * 2  # y coordinate in [-1, 1] range
+    radius = np.sqrt(1 - ys**2)  # circle radius at height y
 
+    theta = angle_step * indices
 
-def random_points_on_circle(axis, radius=1.0, nPoints=100):
-    axis = Vector(axis).normal().array
+    xs = np.cos(theta) * radius
+    zs = np.sin(theta) * radius
 
-    points = np.random.uniform(-radius, radius, 3 * nPoints)
-    points = points.reshape((3, nPoints))
+    return np.vstack([xs, ys, zs])
 
-    proj = np.dot(axis, points)
-    proj = np.dot(axis[:, np.newaxis], proj[np.newaxis, :])
 
-    points -= proj
+def rotation_matrix(axis: Sequence[float], theta: float) -> npt.NDArray[float]:
+    """Return the rotation axis of rotation around the axis by theta angle.
 
-    points *= radius / np.sqrt(np.sum(points**2, axis=0))
+    Parameters
+    ----------
+    axis : Sequence[float]
+        A 3D vector defining the rotation axis.
+    theta : float
+        Rotation angle in radians.
 
-    return points
+    Returns
+    -------
+    npt.NDArray[float]
+        A 3x3 rotation matrix.
+    """
+    axis = np.asarray(axis)
+    theta = np.asarray(theta)
+    axis = axis / np.sqrt(np.dot(axis, axis))
+    a = np.cos(theta / 2.0)
+    b, c, d = -axis * np.sin(theta / 2.0)
+    aa, bb, cc, dd = a * a, b * b, c * c, d * d
+    bc, ad, ac, ab, bd, cd = b * c, a * d, a * c, a * b, b * d, c * d
+    return np.array(
+        [
+            [aa + bb - cc - dd, 2 * (bc + ad), 2 * (bd - ac)],
+            [2 * (bc - ad), aa + cc - bb - dd, 2 * (cd + ab)],
+            [2 * (bd + ac), 2 * (cd - ab), aa + dd - bb - cc],
+        ]
+    )
```

Comment:
I think this can be replaced with scipy. 

https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html

I think you can use from_rotvec() and then as_matrix().